web/modules/fluent/src/default.ts

```diff
@@ -0,0 +1,73 @@
+import { dedupe, Pref } from "@wikijump/util"
+import {
+  Locale,
+  type FluentData,
+  type UnitFormatOptions,
+  type UnitString
+} from "./locale-bundle"
+import { FALLBACK_LOCALE } from "./locales"
+
+const navigatorLocales = dedupe([navigator.language, ...navigator.languages])
+
+const initialLocale = Pref.get("locale", navigatorLocales[0] ?? FALLBACK_LOCALE)
+const fallbackLocales = [...navigatorLocales]
+
+// avoid our initial locale also being in our fallbacks
+if (fallbackLocales[0] === initialLocale) fallbackLocales.shift()
+
+// push fallback locale at the end if we don't have it anywhere
+if (initialLocale !== FALLBACK_LOCALE && !fallbackLocales.includes(FALLBACK_LOCALE)) {
+  fallbackLocales.push(FALLBACK_LOCALE)
+}
+
+// TODO: can this be made reactive like svelte-i18n?
```

Comment:
`svelte-i18n` has this feature where if you change your language it'll update every string on the fly. That's because it's formatting functions are all observables.

Although, now that I think about it, since we have serverside views, we'd have to refresh the page anyways, so I don't think the reactivity is worth it.

Reply:
Yeah reactivity is nice but it doesn't seem particularly worthwhile here.